### PR TITLE
kvutils | sandbox: Wrap metric names in a value type.

### PR DESCRIPTION
**Pull request by SamirTalwar-DA**
For safety, and for simplicity when building upon prefixes.

### Pull Request Checklist

- [x] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/master/CONTRIBUTING.md)
- [x] Include appropriate tests
- [x] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [x] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

File: ledger/sandbox/src/main/scala/com/digitalasset/platform/apiserver/MetricsNaming.scala [hunk 1]
```diff
@@ -34,24 +34,26 @@ private[apiserver] object MetricsNaming {
   }
 
   // Turns a camelCased string into a snake_cased one
-  val camelCaseToSnakeCase: String => String =
+  private[apiserver] val camelCaseToSnakeCase: String => String =
```
Comment (stefanobaghino-da) [inline ledger/sandbox/src/main/scala/com/digitalasset/platform/apiserver/MetricsNaming.scala]: Why did you re-introduce this?

File: ledger/participant-state-metrics/src/main/scala/com/daml/ledger/participant/state/metrics/MetricName.scala [hunk 2]
```diff
@@ -0,0 +1,26 @@
+// Copyright (c) 2020 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.ledger.participant.state.metrics
+
+import scala.language.implicitConversions
+
+case class MetricName(segments: Vector[String]) extends AnyVal {
```
[the remark this answers is not in the document]
Comment (SamirTalwar-DA) [reply ledger/participant-state-metrics/src/main/scala/com/daml/ledger/participant/state/metrics/MetricName.scala]: Probably doesn't; we're not checking for equality anywhere.

File: ledger/sandbox/src/main/scala/com/digitalasset/platform/apiserver/MetricsNaming.scala (hunk 1 above)
Comment (SamirTalwar-DA) [reply ledger/sandbox/src/main/scala/com/digitalasset/platform/apiserver/MetricsNaming.scala]: Good question!

```suggestion
  val camelCaseToSnakeCase: String => String =
```

**Comment by SamirTalwar-DA** [comment]
Finally passed and now I have merge conflicts. 🙁

File: ledger/participant-state-metrics/src/main/scala/com/daml/ledger/participant/state/metrics/MetricName.scala [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+  def :+(segment: String): MetricName =
+    new MetricName(segments :+ segment)
+
+  override def toString: String =
+    segments.mkString(".")
+}
+
+object MetricName {
+
+  val DAML: MetricName = MetricName("daml")
```
Comment (miklos-da) [inline ledger/participant-state-metrics/src/main/scala/com/daml/ledger/participant/state/metrics/MetricName.scala:18]:
```suggestion
  val Daml: MetricName = MetricName("daml")
```

Comment (SamirTalwar-DA) [reply ledger/participant-state-metrics/src/main/scala/com/daml/ledger/participant/state/metrics/MetricName.scala:18]: Ugh, fine. 🙃

Comment (SamirTalwar-DA) [reply ledger/participant-state-metrics/src/main/scala/com/daml/ledger/participant/state/metrics/MetricName.scala:18]: Whoops, didn't realize that label was still on there. We can fix this in a subsequent PR.

**Pull request merged**